### PR TITLE
chore: V4 cherry pick

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #

Pulled in change from the last v4 major release.

### CODE REVIEW

File: packages/react-table/src/components/Table/TableTypes.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { DropdownItemProps } from '@patternfly/react-core';
+import { ButtonProps } from '@patternfly/react-core/dist/esm/components/Button';
 import { formatterValueType, ColumnType, RowType, RowKeyType, HeaderType } from './base';
 import { SortByDirection } from './SortColumn';
-import { DropdownDirection, DropdownPosition } from '@patternfly/react-core/dist/esm/deprecated/components';
+import { DropdownItemProps } from '@patternfly/react-core/dist/esm/components/Dropdown';
+import { DropdownDirection, DropdownPosition } from '@patternfly/react-core/dist/esm/deprecated/components/Dropdown';
```
Comment (kmcfaul) [inline packages/react-table/src/components/Table/TableTypes.tsx:5]: We may want to make a note to clean up the remaining usages of the old dropdown. Looks like it's still referenced in `IColumn` here and `TdActionsType` in `base/types`.

**Review by kmcfaul — APPROVED** [review]
Minor note about deprecated Dropdown usage but not a blocker. Everything else looks good to me

File: packages/react-table/src/components/Table/TableTypes.tsx (hunk 1 above)
Comment (tlabaj) [reply packages/react-table/src/components/Table/TableTypes.tsx:5]: @thatblindgeye is this part of your PR for updating demos to use the new dropdown?

File: packages/react-table/src/components/TableComposable/examples/Table.md [hunk 2]
```diff
@@ -332,11 +332,30 @@ To make a column sticky, wrap `TableComposable` with `InnerScrollContainer` and
...
+- add `isStickyColumn` to all columns that should be sticky
+- add `hasRightBorder` to the rightmost sticky column
+- add `stickyLeftOffset` to each sticky column after the first, with a value that equals the combined width - set by `stickyMindWidth` - of the previous sticky columns
+
+To prevent the default text wrapping behavior and allow horizontal scrolling, all `Th` or `Td` cells should also have the `modifier="nowrap"` property.
 
 ```ts file="TableMultipleStickyColumns.tsx"
 ```
 
+### Composable: Multiple right-aligned sticky columns
```
Comment (jenny-s51) [inline packages/react-table/src/components/TableComposable/examples/Table.md:345]: For consistency

```suggestion
### Multiple right-aligned sticky columns
```

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
These updates are mostly looking good! I'm just seeing some bugs in our sticky column examples that may have snuck in with these updates; the sticky column demos that should be left-aligned aren't working. Do you think we should open a separate issue @tlabaj ?

File: packages/react-table/src/components/Table/TableTypes.tsx (hunk 1 above)
Comment (thatblindgeye) [reply packages/react-table/src/components/Table/TableTypes.tsx:5]: Yeah that's something that I should be able to incorporate into #8839

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/TimePicker/__tests__/TimePicker.test.tsx [hunk 3]
```diff
@@ -86,6 +86,27 @@ describe('TimePicker', () => {
         expects: { hour: 23, minutes: 59, seconds: null }
       });
     });
+
+    test('should call onChange when pressing Enter', async () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/components/TimePicker/__tests__/TimePicker.test.tsx:90]: Would we want to skip this test for and update it as a followup?

**Review by thatblindgeye — COMMENTED** [review]
Other than Jenny's comment about the sticky example, I also just had the below comment but otherwise things look good

File: packages/react-core/src/components/TimePicker/__tests__/TimePicker.test.tsx (hunk 3 above)
Comment (tlabaj) [reply packages/react-core/src/components/TimePicker/__tests__/TimePicker.test.tsx:90]: what is the issue so I can note it in my PR.

Comment (thatblindgeye) [reply packages/react-core/src/components/TimePicker/__tests__/TimePicker.test.tsx:90]: It's just expecting the wrong time on GitHub for some reason. I pulled your branch locally and the test passes fine there, so just another time related issue with tests

**Review by thatblindgeye — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Review by jenny-s51 — APPROVED** [review]
LGTM! Noting that sticky table bugs are planned to be resolved by  #8829 👍

**Pull request merged**